README.md

```diff
@@ -1,32 +1,244 @@
+```
+
+Java considers this type of casting a "safe operation", i.e. it can never fail,
+because the subclass `Cat` inherits fields and methods from the superclass `Animal`.
+
+But what if we want to call the `purr()` method on the `Cat` object?
+`Animal` holds fewer fields and methods than the `Cat` class since it is the superclass.
+So even though the `myAnimal` variable references a `Cat` object,
+the `myAnimal` variable is still considered to
+be of type `Animal`, so the fields and methods that are part of the `Cat`
+type are not accessible using the `myAnimal` variable.
+
+For example, this code won't compile since the declared type of `myAnimal`
+is `Animal`, not `Cat`:
+
+```java
+public static void main(String[] args) {
+    Animal myAnimal;
+    myAnimal = new Cat();
+    myAnimal.setAge(5);
+    System.out.println(myAnimal.getAge());
+
+    //compile-time error
+    myAnimal.purr();
+}
+```
+
+
+Why would we want to prevent this? Consider the following example.
+The code randomly assigns an `Animal` or `Cat` to the
+`myAnimal` variable.  While it is safe to call `setAge` and `getAge`
+due to inheritance, it may not be safe to call `purr`.
+
+```java
+public static void main(String[] args) {
+    Animal myAnimal;
+    Random rand = new Random();
+    if (rand.nextBoolean()) {
+        myAnimal = new Cat();
+    }
+    else {
+        myAnimal = new Animal();
+    }
+    myAnimal.setAge(5);
+    System.out.println(myAnimal.getAge());
+
+    //compile-time error
+    myAnimal.purr();
+}
+```
+
+
+### Downcasting or Narrowing
+
+The reverse, or downcasting (also known as narrowing) is not always a "safe
+operation." We can think about it like this: every cat is an animal, but not
+every animal is a cat. The latter is what we are dealing with when we attempt to
+downcast an object.
+
+We have to make sure that the object we are downcasting is compatible with the
+variable we are trying to assign it to. One quick check we could perform is by
+using the `instanceof` operator. The `instanceof` operator checks to see whether
+an object is an instance of a particular class or not and will return a boolean
+value. If it evaluates to true, then the object is an instance of the class.
+
+```java
+Animal myAnimal = new Cat();
+
+if (myAnimal instanceof Cat) {
+    System.out.println("I'm a cat!");
+} else if (myAnimal instanceof Animal) {
+    System.out.println("I'm an animal!");
+} else {
+    System.out.println("I'm something else!");
+}
+```
+
+This produces the following output:
+
+```plaintext
+I'm a cat!
+```
+
+As we can see, `myAnimal` object is an object of type `Cat`, not of type
+`Animal` That's because there is a difference between the type at compile-time
+and the type at runtime. At compile time, meaning when the code was written, the
+type of the `myAnimal` variable was defined as `Animal`. However, when the
+program runs, a `Cat` data type is assigned to `myAnimal` variable instead. So
```

Comment:
Missing "the" between "to" and `myAnimal`

Reply:
fixed